conan/tools/apple/apple.py

```diff
@@ -82,8 +82,14 @@ class XCRun(object):
     def __init__(self, conanfile, sdk=None):
         """sdk=False will skip the flag
            sdk=None will try to adjust it automatically"""
-        if sdk is None and conanfile.settings:
-            sdk = apple_sdk_name(conanfile.settings)
+
+        if hasattr(conanfile, "settings_target") and conanfile.settings_target is not None:
```

Comment:
So we can assume that if ``settings_target`` exists, this means that we are already in a ``tool_require`` and we always want to refer our ``XCRun()`` to the target context? Aren't there any cases, where it will want just the ``XCRun()`` of the current "build" context? Maybe it should be an optional argument?

Reply:
Yes, that also may happen, I'll add an argument so you can get the info based on the build settings.

Reply:
I've looked into this a bit deeper and I believe that relying on `settings_target` by default should be correct. The information that `xcrun` outputs is "on the build machine, _for_ the target platform" - it's only used in the context of building something from source.

The cases I've seen in Conan Center, call `xcrun` to set up arguments to pass to build tools (so it would impact the  `generate()` and `build()` methods).

The case in https://github.com/conan-io/conan/issues/12260 is actually invoking it in `package_info()`, for a package that is a `tool_require` (likely in a profile), and `xcrun` is called in the `package_info`. So "only used in the context of building something from source" still applies, but now with a level of indirection.

If a `tool_require` A has a dependency on another `tool_require` B such that A is built from source, I believe the `settings_target` should cover this case, by taking the values from the `build` profile (docs [here](https://docs.conan.io/en/latest/systems_cross_building/cross_building.html?highlight=settings_target)).

The only case where I can see where one may need to call `xcrun` from the build context settings, is those recipes that need to invoke two compilers, for example those that first need to bootstrap a tool with the "build" context compiler, and then invoke the compiler that is set up for the host context. There aren't many of these, and in those cases because the "build" contest is always macOS to run on macOS, they may be able to rely on calling the compiler with its default without going through `xcrun` -

TL;DR: always use `settings_target` by default for the overwhelmingly typical case, leave an option to force the build context "just in case"